### ISSUE TITLE
react/no-unused-proptype false positive

### ISSUE THREAD

**Issue by georeith** [yannickcr/eslint-plugin-react#1183]
The following is considered an error for the props `a` and `b`:

```js
import React from 'react';

export default function SomeComponent(props) {
    const callback = () => {
        props.a(props.b);
    };

    const anotherCallback = () => {};

    return (
        <SomeOtherComponent
            name={props.c}
            callback={callback}
        />
    );
}

SomeComponent.propTypes = {
    a: React.PropTypes.func.isRequired,
    b: React.PropTypes.string.isRequired,
    c: React.PropTypes.string.isRequired,
};
```

Removing the line `const anotherCallback = () => {};` fixes the false positive.
The following does not trigger an error:

```js
import React from 'react';

export default function SomeComponent(props) {
    const callback = () => {
        props.a(props.b);
    };

    return (
        <SomeOtherComponent
            name={props.c}
            callback={callback}
        />
    );
}

SomeComponent.propTypes = {
    a: React.PropTypes.func.isRequired,
    b: React.PropTypes.string.isRequired,
    c: React.PropTypes.string.isRequired,
};
```

**Comment by jseminck** [comment]
I have been taking a look at this issue as I wanted to familiarize myself with the codebase a little bit. I managed to reproduce the issue through tests, and found a fix for it. Though I have no idea if the fix is valid or not. It does not break any other tests, though.

I will make a PR with the fix later today or tomorrow. Then anyone more experienced with the code can take a look. Even if my fix is not correct it might help someone else figure out what the issue really is.

**Comment by ljharb** [comment]
Actually, I'm confused here. Class properties do not have var declarations in front of them; `const callback = something` is an error afaik. the proposal only allows `x = y` and `static x = y`.

**Comment by ljharb** [comment]
Thanks, good to start somewhere.

**Comment by ljharb** [comment]
I believe this may be fixed with #1218. Please reopen after the next release if not.
